packages/keymaster/src/keymaster.ts

```diff
@@ -90,20 +91,27 @@ export class Keymaster {
 
   async checkAndPayEnabledNetworks(
     networks: string[],
+    period: number,
     dryrun = false
   ): Promise<void> {
-    for (const net of this.networks.values()) {
-      try {
-        if (networks.length === 0 || networks.includes(net.name)) {
-          await net.checkAndPay(dryrun);
+    const promises = Array.from(this.networks.values()).map(async (net) => {
+      while (true) {
```

Comment:
Is this `while(true)` needed here?

Reply:
nvm I see now

Reply:
Please add a comment here explaining what this `.map` with a `while(true)` is being used for, as it's relatively unintuitive at first glance.

Reply:
Added comment